src/io/flutter/utils/DiagnosticsNode.java

```diff
@@ -0,0 +1,502 @@
+   */
+  public boolean getMissingIfNull() {
+    return getBooleanMember("missingIfNull", false);
+  }
+
+  /**
+   * String representation of exception thrown if accessing the property
+   * [value] threw an exception.
+   */
+  public String exception() {
+    return getStringMember("exception");
+  }
+
+  /**
+   * Whether accessing the property throws an exception.
+   */
+  boolean hasException() {
+    return json.has("exception");
+  }
+
+  /**
+   * String representation of the type of the property [value].
+   *
+   * This is determined from the type argument `T` used to instantiate the
+   * [DiagnosticsProperty] class. This means that the type is available even if
+   * [value] is null, but it also means that the [propertyType] is only as
+   * accurate as the type provided when invoking the constructor.
+   *
+   * Generally, this is only useful for diagnostic tools that should display
+   * null values in a manner consistent with the property type. For example, a
+   * tool might display a null [Color] value as an empty rectangle instead of
+   * the word "null".
+   */
+  public String getPropertyType() {
+    return getStringMember("propertyType");
+  }
+
+  /**
+   * If the [value] of the property equals [defaultValue] the priority [level]
+   * of the property is downgraded to [DiagnosticLevel.fine] as the property
+   * value is uninteresting.
+   *
+   * [defaultValue] has type [T] or is [kNoDefaultValue].
+   */
+  public DiagnosticLevel getDefaultLevel() {
+    return getLevelMember("defaultLevel", DiagnosticLevel.info);
+  }
+
+  /**
+   * Whether the value of the property is a Diagnosticable value itself.
+   * Optionally, properties that are themselves Diagnosticable should be
+   * deplayed as trees of diagnosticable properties and children.
+   *
+   * TODO(jacobr): add helpers to get the properties and children of
+   * this diagnosticable value even if getChildren and getProperties
+   * would return null. This will allow showing nested data for properties
+   * that don't show children by default in other debugging output but
+   * could.
+   */
+  public boolean getIsDiagnosticableValue() {
+    return getBooleanMember("isDiagnosticableValue", false);
+  }
+
+  /**
+   * Service used to retrieve more detailed information about the value of
+   * the property and its children and properties.
+   */
+  private final InspectorService inspectorService;
+
+  /**
+   * JSON describing the diagnostic node.
+   */
+  private final JsonObject json;
+
+  private CompletableFuture<ArrayList<DiagnosticsNode>> children;
+
+  private CompletableFuture<ArrayList<DiagnosticsNode>> properties;
+
+  /**
+   * Reference to the actual Dart DiagnosticsNode object this Java DiagnosticsNode
+   * describes.
+   *
+   * We cannot use observatory protocol InstanceRef ids as they are not adequately long lived.
+   */
+  private String dartDiagnosticId;
+
+  // XXX document
```

Comment:
Possibly migrate to the `TODO(user_name):` convention?

Reply:
My convention is all XXX comments are one I expect to remove before landing.